tests/0150-telemetry_mock.c

```diff
@@ -0,0 +1,450 @@
+             */
+            {.ApiKey           = RD_KAFKAP_PushTelemetry,
+             .broker_id        = 2,
+             .expected_diff_ms = push_interval,
+             .jitter_percent   = 0},
+        };
+        SUB_TEST();
+
+        mcluster = test_mock_cluster_new(2, &bootstraps);
+        rd_kafka_mock_telemetry_set_requested_metrics(mcluster,
+                                                      expected_metrics, 1);
+        rd_kafka_mock_telemetry_set_push_interval(mcluster, push_interval);
+
+        /* Set broker 2 down, to make sure broker 1 is the first preferred
+         * broker. */
+        rd_kafka_mock_broker_set_down(mcluster, 2);
+
+        test_conf_init(&conf, NULL, 30);
+        test_conf_set(conf, "bootstrap.servers", bootstraps);
+        test_conf_set(conf, "debug", "telemetry");
+        // rd_kafka_conf_set_error_cb(conf, test_error_is_not_fatal_cb);
+        test_curr->is_fatal_cb = test_error_is_not_fatal_cb;
+        producer = test_create_handle(RD_KAFKA_PRODUCER, conf);
+
+        /* Poll for enough time that the initial GetTelemetrySubscription can be
+         * sent and the first PushTelemetry request can be scheduled. */
+        test_poll_timeout(producer, 0.5 * push_interval);
+
+        /* Poll for enough time that 2 PushTelemetry requests can be sent. Set
+         * the all brokers up during this time, but the preferred broker (1)
+         * should remain sticky. */
+        rd_kafka_mock_broker_set_up(mcluster, 2);
+        test_poll_timeout(producer, 2 * push_interval);
+
+        /* Set the preferred broker (1) down. */
+        rd_kafka_mock_broker_set_down(mcluster, 1);
+
+        /* Poll for enough time that 1 PushTelemetry request can be sent. */
+        test_poll_timeout(producer, 1.25 * push_interval);
+
+        /* Poll for enough time that 1 PushTelemetry request can be sent.  Set
+         * the all brokers up during this time, but the preferred broker (2)
+         * should remain sticky. */
+        rd_kafka_mock_broker_set_up(mcluster, 1);
+        test_poll_timeout(producer, 1.25 * push_interval);
+
+        requests = rd_kafka_mock_get_requests(mcluster, &request_cnt);
+        test_telemetry_check_protocol_request_times(requests, request_cnt,
+                                                    requests_expected, 5);
+
+        /* Clean up. */
+        test_clear_request_list(requests, request_cnt);
+        rd_kafka_destroy(producer);
+        test_mock_cluster_destroy(mcluster);
+
+        SUB_TEST_PASS();
+}
+
+int main_0150_telemetry_mock(int argc, char **argv) {
+
+        if (test_needs_auth()) {
+                TEST_SKIP("Mock cluster does not support SSL/SASL\n");
+                return 0;
+        }
+
+        do_test_telemetry_get_subscription_push_telemetry();
+
+        do_test_telemetry_empty_subscriptions_list();
+
+        do_test_telemetry_terminating_push();
+
+        do_test_telemetry_preferred_broker_change();
+
```

Comment:
Can we add a TODO here for the remaining cases like the Push telemetry erroring out etc.

Reply:
I'll address this after adding a few more tests too